b9lab-content/5-my-own-chain/03-starport-08-reject-game.md

```diff
@@ -0,0 +1,130 @@
+---
+title: The Reject Game Elements
+order: 10
+description: You reject a game.
+---
+
+# The Reject Game Elements
+
+A natural counterpart to anyone being able to create a game for any two other players, is for these players to be able to reject a game. However, a player should not be allowed to reject a game once they have made their first move. Anyway, as you are already accustomed with Starport, see [Create Message](./03-startport-04-create-message) and [Create Handling](./03-startport-05-create-handling), let's go right to business. It is sufficient to say that to reject a game, a player only needs to specify:
+
+* The id of the game to reject. Let's call the field `idValue`.
+* Nothing else really, as the signer of the message is implicitly the player.
+
+## With Starport
+
+Let's name the message object `RejectGame`, so now you invoke Starport:
+
+```sh
+$ starport scaffold message rejectGame idValue --module checkers
+```
+It has created all the boilerplate for you and has left a single place so that you can write the meat of your code. In `x/checkers/keeper/msg_server_reject_game.go`:
+
+```go [https://github.com/cosmos/b9-checkers-academy-draft/blob/93d048c2b7fbdc26825b41bd043d6203ec9c861c/x/checkers/keeper/msg_server_reject_game.go#L10-L17]
+func (k msgServer) RejectGame(goCtx context.Context, msg *types.MsgRejectGame) (*types.MsgRejectGameResponse, error) {
+    ctx := sdk.UnwrapSDKContext(goCtx)
+
+    // TODO: Handling the message
+    _ = ctx
+
+    return &types.MsgRejectGameResponse{}, nil
+}
+```
+
+## Additional Information
+
+A player cannot reject a game once they've played. However, as of now, you do not know whether a player played or not. To remediate this, you need to add a new field to the `StoredGame`. Let's call it `MoveCount`. So in `proto/checkers/stored_game.proto`:
+
+```proto [https://github.com/cosmos/b9-checkers-academy-draft/blob/329c6d0ae8c1dffa85cd437d0cebb246a827dfb2/proto/checkers/stored_game.proto#L15]
+storedGame := types.StoredGame{
+    ...
+    uint64 moveCount = 7;
+}
+```
+At this point, to have Protobuf recompile the relevant Go files, you may want to run:
```

Comment:
You can simplify this step by using

`starport generate proto-go`

Reply:
Thank you.